File: app/src/main/java/com/bitflaker/lucidsourcekit/database/alarms/updated/daos/StoredAlarmDao.java
package com.bitflaker.lucidsourcekit.database.alarms.updated.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.bitflaker.lucidsourcekit.database.alarms.updated.entities.StoredAlarm;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface StoredAlarmDao {
    @Query("SELECT * FROM StoredAlarm")
    Single<List<StoredAlarm>> getAll();

    @Query("SELECT * FROM StoredAlarm WHERE alarmId IN (:alarmIds)")
    Single<List<StoredAlarm>> getAllById(List<Integer> alarmIds);

    @Query("SELECT * FROM StoredAlarm WHERE alarmId = :alarmId")
    Single<StoredAlarm> getById(int alarmId);

    @Query("UPDATE StoredAlarm SET requestCodeActiveAlarm = :requestCode WHERE alarmId = :storedAlarmId")
    Completable updateRequestCode(long storedAlarmId, int requestCode);

    @Update
    Completable update(StoredAlarm alarm);

    @Insert
    Single<Long> insert(StoredAlarm alarm);

    @Delete
    Completable delete(StoredAlarm alarm);

    @Query("DELETE FROM StoredAlarm")
    Completable deleteAll();

    @Query("UPDATE StoredAlarm SET isAlarmActive = :isAlarmActive WHERE alarmId = :alarmId")
    Completable setActiveState(int alarmId, boolean isAlarmActive);

    @Query("DELETE FROM StoredAlarm WHERE alarmId IN (:alarmsToDelete)")
    Completable deleteAllById(List<Integer> alarmsToDelete);
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/database/alarms/updated/daos/StoredAlarmDao.kt
package com.bitflaker.lucidsourcekit.database.alarms.updated.daos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.bitflaker.lucidsourcekit.database.alarms.updated.entities.StoredAlarm

@Dao
interface StoredAlarmDao {
    @Query("SELECT * FROM StoredAlarm")
    suspend fun getAll(): List<StoredAlarm>

    @Query("SELECT * FROM StoredAlarm WHERE requestCodeActiveAlarm != -1")
    suspend fun  getAllActive(): List<StoredAlarm>

    @Query("SELECT * FROM StoredAlarm WHERE alarmId IN (:alarmIds)")
    suspend fun getAllById(alarmIds: List<Long>): List<StoredAlarm>

    @Query("SELECT * FROM StoredAlarm WHERE alarmId = :alarmId")
    suspend fun getById(alarmId: Long): StoredAlarm

    @Query("UPDATE StoredAlarm SET requestCodeActiveAlarm = :requestCode WHERE alarmId = :storedAlarmId")
    suspend fun updateRequestCode(storedAlarmId: Long, requestCode: Int)

    @Update
    suspend fun update(alarm: StoredAlarm)

    @Insert
    suspend fun insert(alarm: StoredAlarm): Long

    @Delete
    suspend fun delete(alarm: StoredAlarm)

    @Query("DELETE FROM StoredAlarm")
    suspend fun deleteAll()

    @Query("UPDATE StoredAlarm SET isAlarmActive = :isAlarmActive WHERE alarmId = :alarmId")
    suspend fun setActiveState(alarmId: Long, isAlarmActive: Boolean)
}
